shared/oae/api/oae.api.tenancysearch.js

```diff
@@ -0,0 +1,139 @@
+ */
+
+define(['jquery', 'oae.core', 'underscore', 'select2'], function($, oae, _, select2) {
+
+    let location = $.url(encodeURI(window.location.href));
+    let protocol = location.attr('protocol');
+    let query = location.attr('relative');
+
+    return (function () {
+        'use strict';
+
+        let searchTenancyModule =  {
+            init: function() {
+                this.cacheModule();
+                this.bindEvents();
+                this.render();
+            },
+            cacheModule: function() {
+                this.$module = $('#tenancyForm');
+                this.$input = this.$module.find('#tenancyInput');
+                this.$alert = this.$module.find('#tenancyAlert');
+            },
+            bindEvents: function() {
+                this.$input.on('select2:select', this.tenancySelect.bind(this));
+            },
+            render: function() {
+                this.$input.select2({
+                    dropdownParent: $('#signin-modal'),
+                    ajax: {
+                        url: '/api/search/tenants?limit=2000',
+                        dataType: 'json',
+                        delay: 700,
+                        data: function (params) {
+                            return {
+                                q: params.term, // search term
+                                page: params.page
+                            };
+                        },
+                        error: function (data) {
+                            if (data.status === 0) {
+                                return;
+                            } else if ((data.status === 404 && data.statusText === 'Not Found') || (data.status === 400 && data.statusText === 'Bad Request')) {
+                                searchTenancyModule.$module.addClass('fail');
+                                searchTenancyModule.$input.select2('destroy');
+                            }
+                        },
+                        processResults: function (data, params) {
+                            // parse the results into the format expected by Select2
+                            // since we are using custom formatting functions we do not need to
+                            // alter the remote JSON data, except to indicate that infinite
+                            // scrolling can be used
+                            let tenancyResults = _.map(data.results, (eachResult) => {
+                                return {
+                                    host: eachResult.host,
+                                    displayName: eachResult.displayName,
+                                    id: eachResult.alias,
+                                    code: eachResult.countryCode
+                                };
+                            });
+
+                            tenancyResults = tenancyResults.slice(0, 10);
+                            tenancyResults.sort(function (a, b) {
```

Comment:
We sort results by length of display name? 😕

Reply:
It makes no sense, I know, but that's how the unity.ac works, and I just thought there must have been a reason so I left it the same. Any ideas on how to do this better? maybe sort alphabetically?

Reply:
I think alphabetical would be better... I don't know what the reasoning there was but to me it doesn't really make sense.